core/src/main/java/com/backblaze/b2/json/B2JsonObjectHandler.java

```diff
@@ -162,21 +159,38 @@ protected void initializeImplementation(B2JsonHandlerMap handlerMap) throws B2Js
 
         // Figure out the argument positions for the constructor.
         {
+            // Parse @B2Json.constructor#params into an array
             String paramsWithCommas = annotation.params().replace(" ", "");
-            String [] paramNames = paramsWithCommas.split(",");
-            if (paramNames.length == 1 && paramNames[0].length() == 0) {
-                paramNames = new String [0];
+            String[] annotationParamNames = paramsWithCommas.split(",");
+            if (annotationParamNames.length == 1 && annotationParamNames[0].length() == 0) {
+                annotationParamNames = null;
             }
 
-            final int constructorParamCount = fields.length + numberOfVersionParams;
-            if (paramNames.length != constructorParamCount) {
-                throw new IllegalArgumentException(clazz.getName() + " constructor does not have the right number of parameters");
+            // Verify that, if present, the number of params specified in the annotation is correct
+            final int expectedParamCount = fields.length + numberOfVersionParams;
+            if (annotationParamNames != null && annotationParamNames.length != expectedParamCount) {
+                throw new B2JsonException(clazz.getName() + " constructor's @B2Json.constructor annotation does not have the right number of params.");
```

Comment:
I think I like the change to `B2JsonException` but maybe @brianb-backblaze had a reason to use `IllegalArgumentException` originally...?

Reply:
I'm fine with the change.